src/serverMain/kotlin/com/studo/campusqr/database/InitialSetup.kt

```diff
@@ -68,19 +77,86 @@ suspend fun initialDatabaseSetup() {
       insert("ldapUserDisablingIntervalMinutes", 24 * 60)
     }
 
-    // Create root user
+
+    // Create root or demo user
     with(getCollection<BackendUser>()) {
-      val rootUser = BackendUser().apply {
-        _id = "rootUser"
-        email = "admin@example.org"
-        passwordHash = Algorithm.hashPassword("admin")
-        name = "Root User"
-        createdDate = Date()
-        createdBy = _id
-        type = UserType.ADMIN
+      if (DEMO_MODE == 1) {
+        // Delete root user so there is no other way to log in
+        // TODO decide if we should just drop all users if in demo mode?
+        deleteOne(BackendUser::_id equal "rootUser")
+        val user =
+          BackendUser().apply {
+            _id = "demoUser"
+            email = "demo@example.org"
+            passwordHash = Algorithm.hashPassword("demo")
+            name = "Demo User"
+            createdDate = now
+            createdBy = _id
+            type = UserType.MODERATOR
+          }
+        insertOne(user, upsert = true)
+      } else {
+        deleteOne(BackendUser::_id equal "demoUser")
+        if (count() == 0L) {
+          val user =
+            BackendUser().apply {
+              _id = "rootUser"
+              email = "admin@example.org"
+              passwordHash = Algorithm.hashPassword("admin")
+              name = "Root User"
+              createdDate = now
+              createdBy = _id
+              type = UserType.ADMIN
+            }
+          insertOne(user, onDuplicateKey = { })
+        }
+      }
+
+      if (DEMO_MODE == 1) {
+        resetLocationsPeriodic()
+      }
+    }
+  }
+}
+
+// Resets locations db on start and every 24 hours
+suspend fun resetLocationsPeriodic() = serverScope.launch {
```

Comment:
As mentioned above: We should not only reset some data (not only locations but all), but also provide some demo values.